### ISSUE TITLE
spatial perturbation analysis

### ISSUE THREAD

**Issue by grst** [scverse/squidpy#784]
Most functions in squidpy seem to be centered around analyzing a single sample. 
For me, in the context of clinical trials, it would be important to compare between sample groups. The usual variables of interest are
 * comparisons of samples over time (pre-treatment vs. on-treatment), i.e. "how does our drug influence the tissue?" and
 * comparisons of responders vs. non-responders, i.e. "What could explain treatment failure?"

A few things that came to my mind
 * Differential ligand/receptor interactions between groups
 * Differential neighborhood analysis (e.g. tumor more infiltrated after treatment?)
 * differential spatial distance (e.g. tumor and certain immune cells closer together in responders?)

Would be great to have dedicated functions for this, and happy about further ideas! 

CC @Zethson, because this is basically "spatial pertpy"

**Comment by Zethson** [comment]
I'd like to CC @AnnaChristina here who's also been working on this AFAIK.

**Comment by grst** [comment]
@timtreis, here is one for "spatial perturbation analysis already"

**Comment by Zethson** [comment]
@grst we're discussing this now and might get back with an update here

**Comment by giovp** [comment]
@Zethson we discussed this before going on holiday, we have a zulip channel for that, I'll add you

**Comment by grst** [comment]
[cellcharter](https://github.com/CSOgroup/cellcharter/tree/main) already has a function for differential neighborhood enrichment that does statistics on the sample level: https://github.com/CSOgroup/cellcharter/blob/396c415f706e46ce5d9b82df062d0b6509aa526e/src/cellcharter/gr/_nhood.py#L136

**Comment by FrancescaDr** [comment]
I agree with @grst that it would be really useful to have differential versions for some of the functions. I am not sure what the status on this is but I just worked with the neighborhood enrichment tests and did not find a possibility to differentiate between conditions. 

Something fairly simple and straightforward could be to add a `condition` key 1) calculate the neighborhood enrichment per `condition`, 2) plot a neighborhood enrichment plot per condition and potentially 3) add a plot with the log-fold change to identify major differences.

**Comment by grst** [comment]
Just saw that the outcome of the owkin hackathon was never mentioned here. During the hackathon, we built a proof of concept for two functions `aggregate_by_node` and `aggregate_by_edge` that allow to extract spatial features per sample into a matrix that can be further analyzed with standard methos (e.g. linear model). 

The idea behind this functions is that for each sample and cell-type/cluster/... we generate one value based on a metric. Examples could be:
 • by node: Average Shannon entropy of neighboring cell-type fractions for all cells of a given type
 • by edge: How often does an edge between two niches occur, relative to all edges in the graph
This results in a sample:cell_type x features matrix that can be analyzed like a gene expression matrix. 

<img width="1759" height="930" alt="Image" src="https://github.com/user-attachments/assets/77c609f0-cbcf-4600-8aac-a37f575935d3" />

<img width="1856" height="926" alt="Image" src="https://github.com/user-attachments/assets/61aef7e3-3ee1-4b57-a4e4-c187ba2f5097" />

(see [presentation](https://docs.google.com/presentation/d/1VRPXuMddOYS_-4IsDOgh0UFIFB_rENiGoJsoIX__zuU/edit?slide=id.g3417a2f16c8_0_260&pli=1#slide=id.g3417a2f16c8_0_260)

The hackathon code would have to be reimplemented more efficiently to make it into squidpy, but I still like the overall approach.
